### ISSUE TITLE
Need a better translation for tuple range errors

### ISSUE THREAD

**Issue by jrdn91** [mattpocock/ts-error-translator#107]
Maybe this is a limitation in how typescript emits errors for range errors on a tuple type but I just ran into this and had to fix a package that was expecting a tuple of two strings but the library actually takes an array of numbers instead. This was the type

```
colors?: [string, string];
```

And when I had defined the prop like this

```
colors={[secondaryColor[500], secondaryColor[400], secondaryColor[200]]}
```

I get a few Typescript errors that don't properly describe the problem

This is one that once I realized what was going on makes sense

`Type 'string' is not assignable to type 'undefined'.`

Which means it expected everything after the first two props to be undefined and I gave it a string

This error makes more sense but still could be more descriptive

`Type 'number[]' is not assignable to type '[string, string]'.`

Basically it would be great if a more descriptive error was shown for a range error related to a tuple type, something like...

"Expected a tuple of [string, string] but you gave [string, string, string]"

Or

"Tuple type expected two arguments but received three"

**Comment by olasunkanmi-SE** [comment]
Hello, @jrdn91 I think `Type 'string' is not assignable to type 'undefined'.` gives a broad understanding of the issue at hand. Your sample `Type 'number[]' is not assignable to type '[string, string]'` won't give a generic answer but would rather change in accordance to the size of the Tuple. What if you have 6 objects in the tuple? Should it response be `[string, string.....]` until the fifth one?
